shardingsphere-infra/shardingsphere-infra-executor/src/main/java/org/apache/shardingsphere/infra/executor/sql/execute/result/query/type/convert/AbstractJDBCTypeHandlerFactory.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shardingsphere.infra.executor.sql.execute.result.query.type.convert;
+
+import com.google.common.base.Preconditions;
+
+import java.lang.reflect.Field;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.sql.Types;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+
+import org.apache.shardingsphere.infra.exception.ShardingSphereException;
+import org.apache.shardingsphere.spi.ShardingSphereServiceLoader;
+
+/**
+ * JDBC type convert factory class.
+ * @author Sirmin
+ */
+@Slf4j
+public abstract class AbstractJDBCTypeHandlerFactory {
+
+    /**
+     * Define a data structure that holds TypeHandler.
+     * JDBC TYPE (int)
+     *      SIGNED (boolean)
+     *          TypeHandler list(by prioritization)
+     */
+    static final Map<Integer, Map<Boolean, List<TypeHandler<?>>>> TYPE_HANDLER_MAP = new HashMap<>();
+
+    static {
+        loadTypeHandlers();
+    }
+
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    private static synchronized void loadTypeHandlers() {
```

Comment:
`loadTypeHandlers` looks so complex to me. Is it possible to load typeHandlers more easily?